### PR TITLE
Add SEM function, test, and optimize loop functions

### PR DESCRIPTION
**Pull request by AnyaWMa**

### CODE REVIEW

**Comment by richford** [comment]
This PR has .js files in the lib folder but I don't think they should've been added because they are in the `.gitignore` file. Did you by any chance add them with `--force`? In either case, would you mind removing the `lib` folder from the commits?

**Comment by richford** [comment]
No need to rewrite history to remove them completely from the repo. But I don't think we want to commit them.

**Comment by AnyaWMa** [comment]
@richford Thanks! i have removed lib folder

File: src/index.ts [hunk 1]
```diff
@@ -74,13 +87,13 @@ export const estimateAbility = (
...
     let nf = 0;
-    for (let i = 0; i < prior.length; i++) {
-      const theta = prior[i][0];
-      const probability = prior[i][1];
-      const like = likelihood(theta);
+    let theta, probability, like;
+    prior.forEach((e) => {
+      theta = e[0];
+      probability = e[1];
+      like = likelihood(theta);
```
Comment (richford) [inline src/index.ts:94]:
```suggestion
    prior.forEach(([theta, probability]) => {
      const like = likelihood(theta);
```

I'd suggest moving the declarations for theta, probability, and like inside of the `forEach` function to limit their scope.

**Review by richford — COMMENTED** [review]
Thanks @AnyaWMa! I made one more comment about variable scope in one of you forEach methods. Then I think this is good to go!

**Comment by AnyaWMa** [comment]
That looks much cleaner!! Thank you!!

**Pull request merged**